Shopify/cli
# [Feature] Added Remix as an option to the create-app command

### WHY are these changes introduced?
We want partners to be able to create a Remix app from the CLI.  

### WHAT is this pull request doing?

Adds Remix to the list of templates when running create-app.

2 important points:

1. I haven't made it the default option yet because for the initial release the `@shopify/shopify-app-remix` will be a release candidate only.  For editions I think this will become the default option.
2. This will fail right now because we haven't moved `@shopify/shopify-app-remix` into it's own package just yet (See the [README here](https://github.com/Shopify/shopify-app-template-remix/blob/main/README.md#set-up) and screenshot below for more context) and the screenshot below.  Once we release `@shopify/shopify-app-remix` and update the template this will work.  We do not want to merge this until then.

<img width="1409" alt="Screenshot 2023-06-14 at 10 45 31 AM" src="https://github.com/Shopify/cli/assets/690791/eb372fbb-9f46-4a83-9e17-a480706e4327">

### How to test your changes?

1. Run `pnpm create-app`
2. See that the Remix option is there
3. See that it fails (due to point 2 above)

### Post-release steps

None

### Measuring impact

How do we know this change was effective? Please choose one:


Please confirm this is the case.

### Checklist
